.github/workflows/pa11y-accessibility.yml

```diff
@@ -0,0 +1,58 @@
+name: pa11y-accessibility-check
+on:
+  pull_request:
+    branches: [main]
+jobs:
+  pa11y:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.x
+      - name: Install dependencies
+        run: npm install
+      - name: Build the app
+        run: npm run build
+      - name: Install Pa11y
+        run: npm install -g pa11y
+      - name: Install Chromium dependencies (for headless env)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+          wget \
+          curl \
+          ca-certificates \
+          fonts-liberation \
+          libappindicator3-1 \
+          libatk-bridge2.0-0 \
+          libatk1.0-0 \
+          libcups2 \
+          libdbus-1-3 \
+          libgdk-pixbuf2.0-0 \
+          libnspr4 \
+          libnss3 \
+          libx11-xcb1 \
+          libxcomposite1 \
+          libxdamage1 \
+          libxrandr2 \
+          xdg-utils
+      - name: Start VITE server
+        run: |
+          nohup npm run preview &
+          sleep 10
+      - name: Run pa11y
+        run: |
+          output=$(npx pa11y http://localhost:4173 --config=pa11y-config.json || true)
+          echo "$output" > pa11y_output.txt
+          cat pa11y_output.txt
+      - name: Check for accessibility violations and fail the workflow if found
+        run: |
+          if grep -q "Error:" pa11y_output.txt; then
+            echo "Accessibility violations found!"
+            exit 1
+          else
+            echo "No accessibility violations found!"
+          fi
```

Comment:
In this section, we're putting all of pa11y's errors into a temporary text file that we then parse (with `grep`) to pick up and notify us of errors.